### ISSUE TITLE
[no-developer-preview] False positives on afterNextRender (now out of developer preview)

### ISSUE THREAD

**Issue by rdamazio** [angular-eslint/angular-eslint#2534]
**Description and reproduction of the issue**

YOUR LINK HERE: https://stackblitz.com/edit/angular-eslint-afternext

**Versions**

```sh
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 20.0.3
Node: 20.19.1
Package Manager: npm 10.8.2
OS: linux x64

Angular: 20.0.4
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.2000.3
@angular-devkit/build-angular   20.0.3
@angular-devkit/core            20.0.3
@angular-devkit/schematics      20.0.3
@angular/cli                    20.0.3
@schematics/angular             20.0.3
rxjs                            7.8.2
typescript                      5.8.3
zone.js                         0.15.1

```

NOTE: As of Angular 20, afterNextRender and a few others are out of developer preview, yet it keeps being reported by this rule.

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
This needs to be reported to the Angular Team, it is marked as developerPreview by them still: https://github.com/angular/angular/blob/900cd37f68c0beca0f19ccba2e3ccd4c34bdab91/packages/core/src/render3/after_render/hooks.ts#L311

The rule is behaving correctly

**Comment by rdamazio** [comment]
Well, not quite - the overload of the function that I'm actually calling (passing a function instead of a spec) is listed as a public API, just a few lines after what you pointed to:

https://github.com/angular/angular/blob/fa5ae9228d2e9e7d3babbffb26a6816d0b92ff2b/packages/core/src/render3/after_render/hooks.ts#L374

so depending on the parameters passed to that function, it's developer preview or not 😦

**Comment by JamesHenry** [comment]
Oh wow, interesting, @json-derulo please can you see if you can fix such a case? (Even if they fix this particular one by updating the JSDoc)

**Comment by JamesHenry** [comment]
I'm tempted to say this was a one-off mistake by the Angular Team and that in practice they have no intention of doing partial rollouts in developer preview via function signatures, so I'll close this one for now and we can revisit if it happens again
